### PR TITLE
Sort assets by modified date

### PR DESCRIPTION
**Pull request by MrFlashAccount**
### Pull Request Description

Closes: https://github.com/enso-org/cloud-v2/issues/1611, https://github.com/enso-org/cloud-v2/issues/1602

This PR refactors the `EditableSpan` component and improve ergonomics of editing file names.

Now:
1. It displays loading state
2. It shows error if something goes wrong
3. It validates the name
4. Changed behavior of comparing asset names: now it's case insensitive and compare across all assets

Also, this PR changes the default order of assets in Cloud backend to match the order with the Local one

This PR fixes the tooltip when a project being executed by another user

https://github.com/user-attachments/assets/83a0cfd6-b090-497c-8171-729c372b00c8

https://github.com/user-attachments/assets/fcf12082-04a3-4fd3-821e-20519343913a

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [ ] The documentation has been updated, if necessary.
- [ ] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [ ] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      [TypeScript](https://github.com/enso-org/enso/blob/develop/docs/style-guide/typescript.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- [ ] Unit tests have been written where possible.
- [ ] If meaningful changes were made to logic or tests affecting Enso Cloud integration in the libraries, 
      or the Snowflake database integration, a run of the [Extra Tests](https://github.com/enso-org/enso/actions/workflows/extra-nightly-tests.yml) has been scheduled.
  - If applicable, it is suggested to paste a link to a successful run of the Extra Tests.

### CODE REVIEW

File: app/gui/src/dashboard/App.tsx [hunk 1]
```diff
@@ -519,40 +518,38 @@ function AppRouter(props: AppRouterProps) {
   )
 
   return (
-    <FeatureFlagsProvider>
```
Comment (MrFlashAccount) [inline app/gui/src/dashboard/App.tsx:545]: Here I only removed the `FeatureFlagsProvider`

**Review by 4e6 — APPROVED** [review]
The tools part looks good to me

File: app/gui/src/dashboard/providers/FeatureFlagsProvider.tsx [hunk 2]
```diff
@@ -4,55 +4,82 @@
...
+  readonly setFeatureFlags: <Key extends keyof FeatureFlags>(
     key: Key,
-    value: FeatureFlags['featureFlags'][Key],
+    value: FeatureFlags[Key],
   ) => void
 }
 
-const flagsStore = createStore<FeatureFlags>()(
+const flagsStore = createStore<FeatureFlagsStore>()(
   persist(
```
Comment (MrFlashAccount) [inline app/gui/src/dashboard/providers/FeatureFlagsProvider.tsx:33]: @somebody1234 I found that the `zustand` has native support for persisting state, wdyt?

File: app/gui/src/dashboard/providers/TextProvider.tsx [hunk 3]
```diff
@@ -70,18 +68,12 @@ export default function TextProvider(props: TextProviderProps) {
 /** Exposes a property to get localized text, and get and set the current language. */
 export function useText() {
   const { language, setLanguage, locale } = React.useContext(TextContext)
-  const localizedText = text.TEXTS[language]
+
+  const localizedText = text.getDictionary(language)
 
   const getText = React.useCallback<GetText>(
```
Comment (MrFlashAccount) [inline app/gui/src/dashboard/providers/TextProvider.tsx:74]: This function is pulled off the provider, so we can now use it outside of React.

**Comment by PabloBuchu** [comment]
1. Loader after rename sometimes properly waits till title is updated sometimes it sets the name back to old one and refetch updates to new value

https://github.com/user-attachments/assets/11f9db35-7008-4dcd-a725-ad4ea95f0d1a

2. Here you can see the given project is in `Opened` state but it shows as `Closed`
Screenshot 2024-12-19 at 09 33 40

**Comment by MrFlashAccount** [comment]
> Loader after rename sometimes properly waits till title is updated sometimes it sets the name back to old one and refetch updates to new value


It's the backend issue, we discussed it, it requires a fix on the backend

> Here you can see the given project is in Opened state but it shows as Closed

Oh intersting. Will take a look. Guess there are some out of sync state between the `listDirectories` and the `getProjectDetails`

@PabloBuchu is it possible though?

**Comment by MrFlashAccount** [comment]
@PabloBuchu addressed the second issue. Please take a look once you have time 😊

**Review by PabloBuchu — APPROVED** [review]

**Pull request merged**